lib/pymedphys/_dicom/connect/listen.py

```diff
@@ -0,0 +1,207 @@
+            "MR Image Storage": "MR",
+            "Enhanced MR Image Storage": "MRE",
+            "Positron Emission Tomography Image Storage": "PT",
+            "Enhanced PET Image Storage": "PTE",
+            "RT Image Storage": "RI",
+            "RT Dose Storage": "RD",
+            "RT Plan Storage": "RP",
+            "RT Structure Set Storage": "RS",
+            "Computed Radiography Image Storage": "CR",
+            "Ultrasound Image Storage": "US",
+            "Enhanced Ultrasound Image Storage": "USE",
+            "X-Ray Angiographic Image Storage": "XA",
+            "Enhanced XA Image Storage": "XAE",
+            "Nuclear Medicine Image Storage": "NM",
+            "Secondary Capture Image Storage": "SC",
+        }
+
+        try:
+            mode_prefix = mode_prefixes[dataset.SOPClassUID.name]
+        except KeyError:
+            mode_prefix = "UN"
+
+        suid = dataset.SeriesInstanceUID
+        series_dir = self.storage_directory / suid
+        series_dir.mkdir(exist_ok=True)
+
+        filename = "{0!s}.{1!s}".format(mode_prefix, dataset.SOPInstanceUID)
+        filepath = series_dir / filename
+
+        if filepath.exists:
+            # TODO Currently just logging this, since I'm unsure of the desired
+            # behaviour if the file has already been recieved.
+            logging.debug("DICOM file already exists, overwriting")
+
+        context = event.context
+        meta = Dataset()
+        meta.MediaStorageSOPClassUID = dataset.SOPClassUID
+        meta.MediaStorageSOPInstanceUID = dataset.SOPInstanceUID
+        meta.ImplementationClassUID = PYNETDICOM_IMPLEMENTATION_UID
+        meta.TransferSyntaxUID = context.transfer_syntax
+
+        # The following is not mandatory, set for convenience
+        meta.ImplementationVersionName = PYNETDICOM_IMPLEMENTATION_VERSION
+        file_ds = FileDataset(
+            filepath.as_posix(), {}, file_meta=meta, preamble=b"\0" * 128
+        )
+        file_ds.update(dataset)
+        file_ds.is_little_endian = context.transfer_syntax.is_little_endian
+        file_ds.is_implicit_VR = context.transfer_syntax.is_implicit_VR
+
+        status_ds = Dataset()
+        status_ds.Status = 0x0000
+
+        try:
+            # We use `write_like_original=False` to ensure that a compliant
+            #   File Meta Information Header is written
+            file_ds.save_as(filepath.as_posix(), write_like_original=False)
+            status_ds.Status = 0x0000  # Success
+        except IOError:
+            logging.warning("Could not write file to specified directory:")
+            logging.warning("    %s", filepath)
+            logging.warning(
+                "Directory may not exist or you may not have write " "permission"
+            )
+            # Failed - Out of Resources - IOError
+            status_ds.Status = 0xA700
+        except Exception as exception:
+            logging.warning("An error occured saving the Dicom file:")
```

Comment:
suggest using error level as mentioned above

Reply:
If 

(0000,0902) | Error Comment | ErrorComment | LO | 1 | This field contains an application-specific text description of the error detected.
-- | -- | -- | -- | -- | --



is available through pynetdicom in the status_ds, it is highly useful to the SCU (or the people responsible for the SCU).  Which in turn often saves grief for those responsible for the SCP.

Reply:
I'm not the biggest fan of the `except Exception` here. I personally would much rather it stop working, scream at me, and fail fast, than keep on trucking for weeks making me think everything is okay, when instead it's been silently dropping or munging data.

Reply:
I've set the ErrorComment which does seem to be supported by pynetdicom. I've removed that last Exception now Simon so it will crash much more loudly if anything unexpected happens.